lte/gateway/c/core/oai/tasks/mme_app/mme_app_state.cpp

```diff
@@ -60,13 +60,15 @@ hash_table_ts_t* get_mme_ue_state() {
   return MmeNasStateManager::getInstance().get_ue_state_ht();
 }
 
-void put_mme_ue_state(mme_app_desc_t* mme_app_desc_p, imsi64_t imsi64) {
+void put_mme_ue_state(
+    mme_app_desc_t* mme_app_desc_p, imsi64_t imsi64, bool force_ue_write) {
   if (MmeNasStateManager::getInstance().is_persist_state_enabled()) {
     if (imsi64 != INVALID_IMSI64) {
       ue_mm_context_t* ue_context = nullptr;
       ue_context =
           mme_ue_context_exists_imsi(&mme_app_desc_p->mme_ue_contexts, imsi64);
-      if (ue_context && ue_context->mm_state == UE_REGISTERED) {
+      if ((ue_context && !force_ue_write) ||
```

Comment:
isn't it 
if (ue_context && force_ue_write) ?

Reply:
+1

Reply:
good catch, I was using this as reverse (filter_ue_write) but renamed it and forgot to change the condition, will update